lib/src/main/java/com/segment/analytics/kotlin/destinations/consent/ConsentManager.kt

```diff
@@ -0,0 +1,181 @@
+package com.segment.analytics.kotlin.destinations.consent
+
+import com.segment.analytics.kotlin.core.Analytics
+import com.segment.analytics.kotlin.core.BaseEvent
+import com.segment.analytics.kotlin.core.Settings
+import com.segment.analytics.kotlin.core.platform.EventPlugin
+import com.segment.analytics.kotlin.core.platform.Plugin
+import com.segment.analytics.kotlin.core.utilities.toJsonElement
+import com.segment.analytics.kotlin.destinations.consent.Constants.Companion.CATEGORIES_KEY
+import com.segment.analytics.kotlin.destinations.consent.Constants.Companion.CATEGORY_PREFERENCE_KEY
+import com.segment.analytics.kotlin.destinations.consent.Constants.Companion.CONSENT_KEY
+import com.segment.analytics.kotlin.destinations.consent.Constants.Companion.CONSENT_SETTINGS_KEY
+import com.segment.analytics.kotlin.destinations.consent.Constants.Companion.EVENT_SEGMENT_CONSENT_PREFERENCE
+import com.segment.analytics.kotlin.destinations.consent.Constants.Companion.HAS_UNMAPPED_DESTINATIONS_KEY
+import kotlinx.serialization.json.*
+
+import sovran.kotlin.SynchronousStore
+import java.util.concurrent.atomic.AtomicBoolean
+
+
+class ConsentManager(
+    private var store: SynchronousStore,
+    private var consentProvider: ConsentCategoryProvider,
+    private var consentChange: (() -> Unit)? = null
+) : EventPlugin {
+
+
+    override lateinit var analytics: Analytics
+    override val type: Plugin.Type = Plugin.Type.Enrichment
+
+    // Event Queue
+    private var queuedEvents: MutableList<BaseEvent>? = mutableListOf<BaseEvent>()
+
+    // Flag for Event Queue
+    private var started = AtomicBoolean(false)
+
+    init {
+        // IF no consentChanged function passed in, set a default.
+        if (consentChange == null) {
+            consentChange = { analytics.track(EVENT_SEGMENT_CONSENT_PREFERENCE) }
+        }
+    }
+
+    override fun setup(analytics: Analytics) {
+        super.setup(analytics)
+
+        // Empty state
+        store.provide(ConsentState.defaultState)
+    }
+
+    override fun update(settings: Settings, type: Plugin.UpdateType) {
+
+        val state = consentStateFrom(settings)
+        // Update the store
+        store.dispatch(UpdateConsentStateActionFull(state), ConsentState::class)
+
+
+        analytics?.let {
+
+            // Add Segment Destination blocker
+            val segmentDestination = it.find(Constants.SEGMENT_IO_KEY)
+            segmentDestination.let {
+                val existingBlocker =
+                    segmentDestination?.analytics?.find(SegmentConsentBlocker::class)
```

Comment:
why not find from analytics directly?
```suggestion
                    analytics.find(SegmentConsentBlocker::class)
```